src/app/pages/dashboard/types/widget-settings-ref.interface.ts

```diff
@@ -0,0 +1,7 @@
+import { ValidationErrors } from '@angular/forms';
+
+export class WidgetSettingsRef {
+  getData: () => { slot: number; settings: object };
```

Comment:
I don't really see why settings components need to know what slot number they are in.
You can add slot number to `WidgetSettingsRef` in a way that would hide this information or you can just rely on currently selected slot.